src/vm/ceeload.cpp

```diff
@@ -11845,7 +11905,22 @@ static void ProfileDataAllocateScenarioInfo(ProfileEmitter * pEmitter, LPCSTR sc
     // Allocate and initialize the scenario header section
     //
     {
-        LPCWSTR  pCmdLine    = GetCommandLineW();
+        // Get the command line.
+        LPCWSTR pCmdLine;
+
+#if defined(FEATURE_PAL) && defined(FEATURE_PERFTRACING)
+
+        // On UNIX, the PAL doesn't have the command line arguments, so we must build the command line.
+        // EventPipe::SaveCommandLine has already built the command line that we need.
+        pCmdLine = GetManagedCommandLine();
+
+#else // !FEATURE_PAL || !FEATURE_PERFTRACING
```

Comment:
I think that this whole thing can just collapse down to calling GetManagedCommandLine() unconditionally which should have the same behavior as what you have here.